src/app/components/compact-list/compact-list.component.ts

```diff
@@ -0,0 +1,184 @@
+import {Component, Input, Output, ElementRef, Renderer, ViewChild} from "@angular/core";
+import {FormGroup, FormControl, Validators} from "@angular/forms";
+import {ReplaySubject} from "rxjs";
+import {ComponentBase} from "../common/component-base";
+import {ContenteditableDirective} from "./directives/contenteditable.directive";
+import {TagModel} from "./tag.model";
+
+require("./compact-list.component.scss");
+
+export interface ValidationResponse {
+    isValid: boolean,
+    errorMessage: string
+}
+
+@Component({
+    selector: "compact-list",
+    directives: [
+        ContenteditableDirective
+    ],
+    template: `
+       <form [formGroup]="formGroup" class="compact-list-form">
+            <div class="compact-list-wrapper" (click)="onListWrapperClick()">
+                
+                <div class="input-tag-list">
+                    <span *ngFor="let tag of tagList; let i = index; trackBy:trackByIndex"
```

Comment:
I think it uses "object indentity", which is basically object equality. https://angular.io/docs/ts/latest/api/common/index/NgFor-directive.html